crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -0,0 +1,1233 @@
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = function(foo) {
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						};
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = function(foo) {
+							const doBar = function(bar) {
+								return foo + bar;
+							};
+							return foo;
+						};
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo) {
+							const doBar = function(bar) {
+								return foo + bar;
+							};
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo) {
+							function doBar(bar) {
+								return foo + bar;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo = 'foo') {
+							function doBar(bar) {
+								return foo + bar;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							const foo = 'foo';
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo(foo) {
+							function doBar(bar) {
+								function doZaz(zaz) {
+									return foo + bar + zaz;
+								}
+								return bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						for (let foo = 0; foo < 1; foo++) {
+							function doBar(bar) {
+								return bar + foo;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						let foo = 0;
+						function doFoo() {
+							foo = 1;
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = foo => {
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo =
+							foo =>
+							bar =>
+							foo + bar;
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = () => {
+							return bar => bar;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							return bar => bar;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = foo => {
+							const doBar = bar => {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							{
+								const foo = 'foo';
+								function doBar(bar) {
+									return bar + foo;
+								}
+							}
+						}
+					",
+            None,
+        ),
+        //      (
+        //          "
+        //  					function doFoo(foo) {
```

Comment:
Does this test case not pass?

Reply:
ah so this relates to [this](https://github.com/oxc-project/oxc/pull/4948#discussion_r1720820764) - currently this reports an error on it (i think correctly), but eslint-plugin unicorn does not

Reply:
i think it's ok to report an error as there's no reason why, in the following:
```ts
function foo() { 
    { 
         function bar() {}
     }
}
```

`bar` couldn't be moved to the outer scope

Reply:
Can you leave comments above commented or moved tests explaining what behavior changes there are and why?

Reply:
fixed